Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,653 @@
+					:
 					ConverterBaseT<IfcEntityTypesT>(geomSettings, unitConverter),
 					placementConverter(pc),
 					curveConverter(cc)
 				{
 				}
 
+				//! Virtual destructor
 				virtual ~FaceConverterT()
 				{
 				}
 
-				void convertIfcSurface(const std::shared_ptr<typename IfcEntityTypesT::IfcSurface>& surface,
+				/*! \brief Converts \c IfcSurface to a according to subtype.
+					\param		surface	\c IfcSurface entity to be interpreted.
+					\param		pos
+					\param		polylineData
+					\note		The \c IfcSurface subtypes are \c IfcBoundedSurface,  \c IfcElementarySurface and  \c IfcSweptSurface.
+				*/
+
+				void convertIfcSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcSurface>& surface,
 					const carve::math::Matrix& pos,
-					std::shared_ptr<carve::input::PolylineSetData>& polyline_data)
-				{
-					double length_factor = UnitConvert()->getLengthInMeterFactor();
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) {
+
+					// (1/3) IfcBoundedSurface SUBTYPE of IfcSurface
+					if (surface.isOfType<typename IfcEntityTypesT::IfcBoundedSurface>()) {
+						return convertIfcBoundedSurface(surface.as<typename IfcEntityTypesT::IfcBoundedSurface(),
+							pos,
+							polylineData);
+					}
+
+					// (2/3) IfcElementarySurface SUBTYPE of IfcSurface
+					if (surface.isOfType<typename IfcEntityTypesT::IfcElementarySurface>()) {
+						return convertIfcElementarySurface(surface.as<typename IfcEntityTypesT::IfcElementarySurface(),
+							pos,
+							polylineData);
+					}
+
+					// (3/3) IfcSweptSurface SUBTYPE of IfcSurface
+					if (surface.isOfType<typename IfcEntityTypesT::IfcSweptSurface>()) {
+						return convertIfcSweptSurface(surface.as<typename IfcEntityTypesT::IfcSweptSurface(),
+							pos,
+							polylineData);
+					}
+				}
 
-					/*	Faceconverter.h
-						For IFC4x1:
+				//--------------------------------------------------------------------------------------------
+				// IfcBoundedSurface
+				//--------------------------------------------------------------------------------------------
 
-						IfcSurface
+						/*! \brief  Converts \c IfcBoundedSurface by calling the conversion function according to subtype.
+						\param		surface	\c IfcBoundedSurface entity to be interpreted.
+						\param		pos
+						\param		polylineData
+						\note		The \c IfcBoundedSurface subtypes are: \c IfcBSplineSurface, \c IfcCurveBoundedPlane, \c IfcCurveBoundedSurface and \c IfcRectangularTrimmedSurface.
+						*/
 
-							IfcBoundedSurface						http://www.buildingsmart-tech.org/ifc/IFC4x1/final/html/schema/ifcgeometryresource/lexical/ifcboundedsurface.htm
-								IfcBSplineSurface*
-								IfcCurveBoundedPlane
-								IfcCurveBoundedSurface*
-								IfcRectangularTrimmedSurface*
+				void convertIfcBoundedSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcBoundedSurface>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) {

```

Comment:
What @hechth means is:
```suggestion
				std::shared_ptr<carve::input::PolylineSetData> convertIfcBoundedSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcBoundedSurface>& surface,
					const carve::math::Matrix& pos) const throw(...) 
				{
					std::shared_ptr<carve::input::PolylineSetData> polylineData = std::make_shared<carve::input::PolylineSetData>();
					/*...*/
					return polylineData;

```

Reply:
Also, I would prefer, if the starting brace `{` were always on a new line.

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/70b4a05752c28606f9282cce8fa687432c87003f